tutorials/thermal_camera.md

```diff
@@ -0,0 +1,268 @@
+
+  for (auto r = 0; r < _msg.height(); ++r)
+  {
+    // need to figure out the row offset in order to mimic 2D array access
+    // with 1D array indexing
+    auto row_offset = r * _msg.width();
+
+    for (auto c = 0; c < _msg.width(); ++c)
+    {
+      // convert the 16 bit value to Kelvin via the camera's linearResolution
+      auto temp = thermalBuffer[row_offset + c] * linearResolution;
+
+      // do something useful with the temperature (in Kelvin) here
+    }
+  }
+
+  delete[] thermalBuffer;
+}
+
+int main(int argc, char **argv)
+{
+  ignition::transport::Node node;
+  if (!node.Subscribe("/thermal_camera", &OnImage))
+  {
+    std::cerr << "Error subscribing to the thermal camera topic" << std::endl;
+    return -1;
+  }
+
+  ignition::transport::waitForShutdown();
+}
+```
+
+Although most of the code above is described in the comments, let's go over the key points again:
+* The image published by the camera is represented as serialized bytes.
+In order to read the actual temperature data that's stored in the image, we need to convert the data in the image to 16 bit integers.
+* Once the image is stored in an array of 16 bit integers, we'll need to multiply the data in this array by the camera's linear resolution in order to ensure we are working with units of Kelvin.
+Since this thermal camera has a linear resolution of .01 (10mK), and the maximum value that can be stored in an unsigned 16 bit integer is 65535, this means that the maximum temperature that can be recorded by this camera is (65535 * .01) = 655.35 Kelvin / 680.72 F / 360.4 C.
+
+## Thermal Camera Limitations
+
+The thermal camera has a few limitations:
+* When assigning a temperature to a model, that temperature is applied equally across the whole model instead of having a gradient that falls off from the center of a heat source.
+    - The current implementation allows for a temperature to be assigned to a model, but doesn't allow for the location of a heat source in a model to be specified.
+* If one object with a given temperature blocks another object with a given temperature (with respect to the thermal camera's point of view), the blocked object will not be displayed at all, regardless of each object's thickness and temperature difference between them (see the image below for an example).
+A more realistic implementation would have the camera display temperature fluctuations in the closer object if the temperature difference between the two objects is large enough, and if the closer object isn't too thick.
+    - More information about thermal camera behavior can be found [here](https://www.flir.com/discover/cores-components/can-thermal-imaging-see-through-walls/).
+* There are some precision issues in the conversion pipeline of a thermal camera's output (converting temperature readings to grayscale output -> serializing this output -> sharing the serialized image -> de-serializing the shared image output back into the original temperature readings).
```

Comment:
I think the precision loss is mainly from the first step (conversion to normalized values in the shaders), and should not be affected by msg serialization, and serialization so we can remove the last few steps.